### PR TITLE
Fix missing canonical-versioning ruleset in arm lib

### PR DESCRIPTION
**Pull request by timotheeguerin**

### CODE REVIEW

File: packages/typespec-azure-resource-manager/src/linter.ts [hunk 1]
```diff
@@ -64,7 +64,10 @@ export const $linter = defineLinter({
         // TODO: Enable this rule once azure-rest-api-specs repo is ready (issue #3839)
         [`@azure-tools/typespec-azure-resource-manager/${armCommonTypesVersionRule.name}`]: false,
       },
-      extends: ["@azure-tools/typespec-azure-core/all"],
+      extends: [
+        "@azure-tools/typespec-azure-core/all",
+        "@azure-tools/typespec-azure-core/canonical-versioning",
```
Comment (tjprescott) [inline packages/typespec-azure-resource-manager/src/linter.ts:69]: If this already extends the `azure-core/all` ruleset, then why does this rule need to be extended here?  Is "all" not really all?

Comment (timotheeguerin) [reply packages/typespec-azure-resource-manager/src/linter.ts:69]: yes all doesn't include those, this was a mistake to name it `all` we should have had
- `required` (What is required for azure core)
- `all` (Actually everything)
- `<anyoptional set>`

Comment (timotheeguerin) [reply packages/typespec-azure-resource-manager/src/linter.ts:69]: i'll file an issue to consider changing the names

Comment (timotheeguerin) [reply packages/typespec-azure-resource-manager/src/linter.ts:69]: https://github.com/Azure/typespec-azure/issues/330

**Review by markcowl — APPROVED** [review]

**Review by tjprescott — APPROVED** [review]

**Pull request merged**